src/Raven.Server/Documents/ShardedHandlers/ShardedSmugglerHandler.cs

```diff
@@ -60,45 +57,41 @@ public async Task PostExport()
             OperationCancelToken token)
         {
-            var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
-
-            var oldOperateOnType = options.OperateOnTypes;
-
-            var operateOnTypes = options.OperateOnTypes &= ~(DatabaseItemType.DatabaseRecord |
-                                                             DatabaseItemType.Subscriptions |
-                                                             DatabaseItemType.Identities |
-                                                             DatabaseItemType.Indexes |
-                                                             DatabaseItemType.ReplicationHubCertificates);
-
-            blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.OperateOnTypes), operateOnTypes);
-
-            using (var outputStream = GetOutputStream(ResponseBodyStream(), options))
-            using (var exportOutputStream = new GZipStream(outputStream, CompressionMode.Compress))
+            var operationIdList = new List<long>();
+            blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.IsShard), true);
+            await using (var outputStream = GetOutputStream(ResponseBodyStream(), options))
+            await using(var writer = new AsyncBlittableJsonTextWriter(jsonOperationContext, new GZipStream(outputStream, CompressionMode.Compress)))
             {
-                await exportOutputStream.WriteAsync(Encoding.UTF8.GetBytes($"{{  \"BuildVersion\" : {ServerVersion.Build}"));
+                writer.WriteStartObject();
+                writer.WritePropertyName("BuildVersion");
+                writer.WriteInteger(ServerVersion.Build);
                 for (int i = 0; i < ShardedContext.ShardCount; i++)
                 {
-                    if (i == ShardedContext.ShardCount - 1) // Last shard need to bring all server wide information
-                    {
-                        blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.OperateOnTypes), oldOperateOnType);
-                    }
-
                     var cmd = new ShardedStreamCommand(this, async stream =>
                     {
-                        using (var gzipStream = new GZipStream(GetInputStream(stream, options), CompressionMode.Decompress))
+                        await using (var gzipStream = new GZipStream(GetInputStream(stream, options), CompressionMode.Decompress))
                         {
-                            var firstLoop = true;
-                            await gzipStream.CopyToAsync(exportOutputStream);
+                            writer.WriteStream(gzipStream);
                         }
 
                     }, blittableJson);
 
+                    operationIdList.Add(ServerStore.Operations.GetNextOperationId());
```

Comment:
there might be a collision between the id generated by the server and the one generated by the database.
to avoid that maybe we should generate negative id for the server?